### PR TITLE
Fix Utils CI

### PR DESCRIPTION
**Pull request by IlyasMoutawwakil**
# What does this PR do?

Fixes tasks processor ci

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: .github/workflows/test_utils.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04, macos-13]
+        python-version: [3.8, 3.9]
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
```
Comment (echarlaix) [inline .github/workflows/test_utils.yml:24]: why is it needed ?

**Review by echarlaix — APPROVED** [review]

File: .github/workflows/test_utils.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        if: matrix.os == 'ubuntu-20.04'
+        with:
+          tool-cache: true
```
Comment (michaelbenayoun) [inline .github/workflows/test_utils.yml:27]: Why?

File: tests/utils/test_task_processors.py [hunk 3]
```diff
@@ -157,11 +160,15 @@ def _test_load_dataset(
                 path,
                 data_keys=data_keys,
                 only_keep_necessary_columns=only_keep_necessary_columns,
+                load_smallest_split=True,
```
Comment (michaelbenayoun) [inline tests/utils/test_task_processors.py:163]: Nice!

File: .github/workflows/test_utils.yml (hunk 1 above)
Comment (echarlaix) [reply .github/workflows/test_utils.yml:24]: also do you think we can reduce the model / dataset size used during testing https://github.com/huggingface/optimum/blob/main/tests/utils/test_dummpy_input_generators.py#L38 ? (using tiny random models for example)

**Comment by IlyasMoutawwakil** [comment]
still fails after two datasets 🥲

File: .github/workflows/test_utils.yml (hunk 2 above)
Comment (IlyasMoutawwakil) [reply .github/workflows/test_utils.yml]: context https://huggingface.slack.com/archives/C05A8AV5J1M/p1716543338536269?thread_ts=1716404954.641349&cid=C05A8AV5J1M

**Comment by IlyasMoutawwakil** [comment]
reducing the number of samples works (it was the tensors written to disk memory and not the dataset splits)

File: optimum/utils/preprocessing/task_processors_manager.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ class TaskProcessorsManager:
     }
 
     @classmethod
-    def get_task_processor_class_for_task(cls, task: str) -> Type:
+    def get_task_processor_class_for_task(cls, task: str) -> Type["TaskProcessor"]:
```
Comment (mht-sharma) [inline optimum/utils/preprocessing/task_processors_manager.py:38]: Could we maintain the same style as in other funds?
```suggestion
    def get_task_processor_class_for_task(cls, task: str) ->"TaskProcessor":
```

**Review by mht-sharma — APPROVED** [review]
Thanks Ilyas for the fixes

File: optimum/utils/preprocessing/task_processors_manager.py (hunk 4 above)
Comment (IlyasMoutawwakil) [reply optimum/utils/preprocessing/task_processors_manager.py:38]: but one returns a constructor and the other returns an instance

Comment (mht-sharma) [reply optimum/utils/preprocessing/task_processors_manager.py:38]: Got it, thanks for clarifying!

**Pull request merged**